src/table/table-role/grid-navigation.tsx

```diff
@@ -191,69 +192,54 @@ class GridNavigationProcessor {
+      KeyCode.right,
+      KeyCode.pageUp,
+      KeyCode.pageDown,
+      KeyCode.home,
+      KeyCode.end,
+    ];
     const ctrlKey = event.ctrlKey ? 1 : 0;
     const altKey = event.altKey ? 1 : 0;
     const shiftKey = event.shiftKey ? 1 : 0;
     const metaKey = event.metaKey ? 1 : 0;
-    const numModifiersPressed = ctrlKey + altKey + shiftKey + metaKey;
-
-    let key = event.keyCode;
-    if (numModifiersPressed === 1 && event.ctrlKey) {
-      key = -key;
-    } else if (numModifiersPressed) {
+    const modifiersPressed = ctrlKey + altKey + shiftKey + metaKey;
+    const invalidModiferCombination =
+      (modifiersPressed && !event.ctrlKey) ||
+      (event.ctrlKey && event.keyCode !== KeyCode.home && event.keyCode !== KeyCode.end);
+
+    if (
+      invalidModiferCombination ||
+      this.isSuppressed(document.activeElement) ||
+      !this.isRegistered(document.activeElement) ||
+      keys.indexOf(event.keyCode) === -1
+    ) {
       return;
     }
 
     const from = this.focusedCell;
-
-    if (this.isSuppressed(document.activeElement) || !this.isRegistered(document.activeElement)) {
-      return;
-    }
-
-    switch (key) {
-      case KeyCode.up:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: -1, x: 0 });
-
-      case KeyCode.down:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: 1, x: 0 });
-
-      case KeyCode.left:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: 0, x: -1 });
-
-      case KeyCode.right:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: 0, x: 1 });
-
-      case KeyCode.pageUp:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: -this.pageSize, x: 0 });
-
-      case KeyCode.pageDown:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: this.pageSize, x: 0 });
-
-      case KeyCode.home:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: 0, x: -Infinity });
-
-      case KeyCode.end:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: 0, x: Infinity });
-
-      case -KeyCode.home:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: -Infinity, x: -Infinity });
-
-      case -KeyCode.end:
-        event.preventDefault();
-        return this.moveFocusBy(from, { y: Infinity, x: Infinity });
-
-      default:
-        return;
-    }
+    event.preventDefault();
+
+    isEventLike(event) &&
+      handleKey(event, {
+        onBlockStart: () => this.moveFocusBy(from, { y: -1, x: 0 }),
+        onBlockEnd: () => this.moveFocusBy(from, { y: 1, x: 0 }),
+        onInlineStart: () => this.moveFocusBy(from, { y: 0, x: -1 }),
+        onInlineEnd: () => this.moveFocusBy(from, { y: 0, x: 1 }),
+        onPageUp: () => this.moveFocusBy(from, { y: -this.pageSize, x: 0 }),
+        onPageDown: () => this.moveFocusBy(from, { y: this.pageSize, x: 0 }),
+        onHome: () =>
```

Comment:
A non-blocking observation here. On a MacBook there is no Home/End buttons on the keyboard. Instead, you can press Fn+Left/Fn+Right. Do you know if the Home/End are reversed when switching to an RTL language? If not, the interaction feels wrong.

Reply:
My understanding is that Home/End are universal, which can make for a confusing experience on Mac because the Fn+Right will go physically Left in RTL. I double checked and this is the intended behavior.